servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -668,14 +675,16 @@ void RendererSceneRenderRD::_render_buffers_post_process_and_tonemap(const Rende
 		RD::get_singleton()->draw_command_end_label();
 	}
 
-	if (use_fsr) {
-		RD::get_singleton()->draw_command_begin_label("FSR 1.0 Upscale");
+	if (rb.is_valid() && spatial_upscaler) {
+		spatial_upscaler->ensure_context(rb);
+
+		RD::get_singleton()->draw_command_begin_label(spatial_upscaler->get_label());
```

Comment:
@DarioSamo note that in this improved version, there aren't two separate blocks with almost identical code for FSR and MetalFX Spatial. By introducing the `SpatialScaler` base class, we can reuse all this code.